distributed/scheduler.py

```diff
@@ -5091,19 +5091,24 @@ async def restart(self, client=None, timeout=30):
                 stimulus_id=stimulus_id,
             )
 
-        nannies = {addr: ws.nanny for addr, ws in self.workers.items()}
+        nanny_workers: dict[str, str] = {
+            addr: ws.nanny for addr, ws in self.workers.items() if ws.nanny
+        }
 
-        for addr in list(self.workers):
-            try:
-                # Ask the worker to close if it doesn't have a nanny,
-                # otherwise the nanny will kill it anyway
-                await self.remove_worker(
-                    address=addr, close=addr not in nannies, stimulus_id=stimulus_id
-                )
-            except Exception:
-                logger.info(
-                    "Exception while restarting.  This is normal", exc_info=True
-                )
+        # Close non-Nanny workers. We have no way to restart them, so we just let them go,
+        # and assume a deployment system is going to restart them for us.
+        close_results = await asyncio.gather(
+            *(
+                self.remove_worker(address=addr, stimulus_id=stimulus_id)
+                for addr in self.workers
+                if addr not in nanny_workers
+            ),
+            return_exceptions=True,
+        )
+        for r in close_results:
+            if isinstance(r, Exception):
+                # TODO this is probably not, in fact, normal.
+                logger.info("Exception while restarting.  This is normal.", exc_info=r)
```

Comment:
Yes, I suggest to remove this. `Scheduler.remove_worker` does not intentionally raise any exceptions nor are there any obvious transitive exceptions that should be raised and could be handled safely.
Therefore, in this local context I don't see the point about implementing *any* exception handling and would prefer to just reraise (ideally to the client).

I think it's even unnecessary to do the `gather(..., raise_exception=False)` etc. foo. If there is an unexpected exception at this point, the only sane thing to do is to restart the entire cluster (e.g. if there is a transition error) but this needs to be done externally (obviously, since we tried restarting already :)).

Reply:
NVM, we're not using `fail_hard` anywhere on the scheduler right now, so I won't add that. I've removed the error suppression and `raise_exception=False`.